index.js

```diff
@@ -141,6 +141,14 @@ function printMap(val, indent, prevIndent, refs, maxDepth, currentDepth, plugins
 }
 
 function printObject(val, indent, prevIndent, refs, maxDepth, currentDepth, plugins) {
+  if (val.toJSON) {
+    while (val.toJSON) {
```

Comment:
I would probably be extra safe and check `typeof val.toJSON === 'function'`, otherwise LGTM!